features/step_definitions/salt_steps.rb

```diff
@@ -854,12 +843,24 @@
 
 SALT_PACKAGES = "salt salt-minion".freeze
 
-Given(/^no Salt packages are installed on remote "(.*)"$/) do |host|
-  if host == "ssh-minion"
-    $ssh_minion.run("test -e /usr/bin/zypper && zypper --non-interactive remove -y #{SALT_PACKAGES}", false)
-  end
-  if host == "centos"
-    $ceos_minion.run("test -e /usr/bin/yum && yum -y remove #{SALT_PACKAGES}", false)
+Given(/^no Salt packages are installed on "(.*)"$/) do |host|
```

Comment:
use this here for refractoring this function

https://github.com/SUSE/spacewalk-testsuite-base/blob/d3392f6518166c10aafdb4560b6861c268d4cecb/features/step_definitions/command_steps.rb#L14